### ISSUE TITLE
Unable to select geojson object with leaflet lasso

### ISSUE THREAD

**Issue by EricG-Personal** [zakjan/leaflet-lasso#51]
I have a test project at https://github.com/ericg-vue-questions/leaflet-test/tree/feature-lasso-france-behavior

(note the feature-lasso-france-behavior branch)

If I draw a lassoed area around France, the France does not show up in the list of selected objects. No other nation behaves this way.

If I draw a small intersecting area with France, France will show up in the list of selected objects.

I have included in the repo a geojson file containing only France. Just change the line in LeafletTest.vue from:

```
import data from './custom.geo.json';
```

to

```
import data from './france.geo.json';
```

https://user-images.githubusercontent.com/931443/178363473-6f0d4b7d-3516-4e22-b12e-6e0e28a5c7e0.mov

The data comes from https://geojson-maps.ash.ms with the following selections:

![image](https://user-images.githubusercontent.com/931443/178364035-7d4fbbd6-f7de-4284-91fb-cc164a076099.png)

I will continue to work on trying to figure out what is happening, but was hoping you would have some idea why it was failing in this case.

**Comment by zakjan** [comment]
Hi, this happens because France polygon contains French Guiana and Corsica. You can either clean up the polygon to contain France only, or set Lasso selection mode to intersection.

**Comment by EricG-Personal** [comment]
L.control.lasso({ intersect: true }).addTo(mapDiv);

The selection mode is set to intersection. It still fails.

**Comment by zakjan** [comment]
Could you try out cleaning up the polygon to avoid French Guiana then?

**Comment by EricG-Personal** [comment]
That does help matters. However, now that I understand the issue a bit better, I can reproduce a similar issue with circling the State of Hawaii. Placing a lassoed area around Hawaii should count as an intersection with the United States, but does not.

**Comment by zakjan** [comment]
The underlying intersection method is used from `@terraformer/spatial` at https://github.com/terraformer-js/terraformer, could you reproduce it there? There is https://github.com/terraformer-js/terraformer/issues/84 which seems related.

**Comment by eric-g-97477** [comment]
Yes, I am certain that https://github.com/terraformer-js/terraformer/issues/84https://github.com/terraformer-js/terraformer/issues/84 is the same issue I am reporting here. It looks like they will be fixing the problem.

**Comment by eric-g-97477** [comment]
@zakjan I appears the underlying bug in terraformer has been fixed.

Is there anything you need to do to pick that fix up in your project?

**Comment by zakjan** [comment]
@terraformer/spatial dependency updated, released as 2.2.12

**Comment by eric-g-97477** [comment]
hummm...

I created a code pen at https://codepen.io/ericg_off/pen/oNqqNxR which is using 2.2.12, but the problem persists.

Can you verify? 

Perhaps I was mistaken about terraformer or there is still a little work to do in this plugin.

**Comment by eric-g-97477** [comment]
I created another codepen https://codepen.io/ericg_off/pen/xxWWGOX which tests terraformer itself and can confirm that it is working. So, I am not sure why lasso would not be at this point.

**Comment by zakjan** [comment]
Ok, could you explore what's the actual cause, please? I'm sorry, but my availability is limited these days.

**Comment by eric-g-97477** [comment]
I should be able to find some time to do that. Make take my awhile as well.
